.github/workflows/checks.yml

```diff
@@ -15,19 +15,69 @@ jobs:
         ]
 
     runs-on: ${{ matrix.os }}
+    # TODO: (#310) Remove when migrating to macOS arm64 runner
+    env:
+      V1_COMPILER_BUILD_PATH: ${{ github.workspace }}/cairo1/target/release
+      V2_COMPILER_BUILD_PATH: ${{ github.workspace }}/cairo2/target/release
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
-      - uses: actions/setup-java@v3
+
+      # TODO: (#310) Remove when migrating to macOS arm64 runner
+      - name: Clone Cairo repos (macOS only)
+        if: matrix.os == 'macos-latest'
+        run: |
+          git clone -b v1.1.1 https://github.com/starkware-libs/cairo.git cairo1
+          git clone -b v2.2.0 https://github.com/starkware-libs/cairo.git cairo2
+
+      - name: Cache Cairo compiler v1 binaries (macOS only)
+        if: matrix.os == 'macos-latest'
+        uses: actions/cache@v3
+        with:
+          path: cairo1/target/release
+          key: ${{ runner.os }}-cairo1-target-release-${{ hashFiles('cairo1/Cargo.lock') }}
+
+      - name: Cache Cairo compiler v2 binaries (macOS only)
+        if: matrix.os == 'macos-latest'
+        uses: actions/cache@v3
+        with:
+          path: cairo2/target/release
+          key: ${{ runner.os }}-cairo2-target-release-${{ hashFiles('cairo2/Cargo.lock') }}
+
+      - name: Build Cairo compiler binaries (macOS only)
+        if: matrix.os == 'macos-latest'
+        run: |
+          if [[ ! -d "$V1_COMPILER_BUILD_PATH" || ! -d "$V2_COMPILER_BUILD_PATH" ]]; then
+            curl -sSf https://sh.rustup.rs | sh -s -- -y
+            if [[ ! -d "$V1_COMPILER_BUILD_PATH" ]]; then
+               cd ${{ github.workspace }}/cairo1 
```

Comment:
I guess it's fine and it works, but I'd encourage you to use `pushd` and `popd` instead